contents/javascript/Javascript-framework-Redux.md

```diff
@@ -0,0 +1,231 @@
+
+![Redux Devtool](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)
+
+<center>
+
+_Redux DevTools_ <sup>[image source](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)</sup>
+</center>
+
+3. **Easier testing**<br/>
+Using Redux makes for testable code. Since the reducers used to change the state are pure functions, it is easy make tests. Writing small, pure and isolated reducer functions in Redux, means we write small and isolated tests. 
+
+4. **Allows for server-side rendering**<br/>
+Redux is one way to fulfill server-side rendering. This can be acheived by sending the store created on the server side in response to the client's server request. Information from the store then can be used for the initial render of the application.
+
+## Why Not to Use Redux?
+
+While Redux has its benefits, you should not add Redux for every application you make.
+
+In general, it would be only apt to use Redux is when:
+
+1. You have reasonable amount of data changing over time
+1. You need a single source of truth
+1. You find keeping everything in a top-level parent component is not enough
+
+Another good rule of thumb is to figure out if you have any problems managing state in your application without Redux. If you don't have any, then you don't really need Redux.  
+
+As Dan Abramov, one of the creators of Redux, says with regards to [React]({{baseUrl}}/contents/javascript/Javascript-framework-React.html), a view library very commonly used with Redux:
+
+<box seamless>
+I would like to amend this: don't use Redux until you have problems with vanilla React.
+</box>
+
+There are many other sites and articles speaking of why you shouldn't or when you shouldn't use Redux. If you are interested, the following resources in this [section](({{baseUrl}}/contents/javascript/javascript-framework-redux#more-articles-on-when-to-use-redux)) provide more information and thoughts on when Redux is meant to be used.
+
+## Getting Started with Redux
+
+The [official Redux website]((https://redux.js.org/)) is a great place to get started. It includes:
+
+- [Installation instructions](https://redux.js.org/introduction/getting-started/)
+- A suggested list of resources encompassing [Basic](https://redux.js.org/introduction/getting-started/#just-the-basics) and [Intermediate](https://redux.js.org/introduction/getting-started/#intermediate-concepts) concepts.
+- A [basic](https://redux.js.org/basics/basic-tutorial) and [advanced](https://redux.js.org/advanced/advanced-tutorial) tutorial.
+
+This article is based on the the sections [Core Concepts](https://redux.js.org/introduction/core-concepts) and [Three Principles](https://redux.js.org/introduction/three-principles) of the introduction, so you may find the sections familiar and be able skim through those.
```

Comment:
`It may be good to reinforce some of the key concepts with the basic tutorial first, then ...`?